### PR TITLE
Support CountWorkflowExecutions for SQL DB

### PR DESCRIPTION
**Pull request by rodrigozhou**
**What changed?**
Implement support for `CountWorkflowExecutions` for advanced visibility with SQL DB.

**Why?**
Advanced visibility with SQL DB.

**How did you test it?**
Enabled `TestCountWorkflow` functional test.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/persistence/visibility/store/sql/query_converter_sqlite.go [hunk 1]
```diff
@@ -307,3 +310,26 @@ func (c *sqliteQueryConverter) buildFtsSelectStmt(
...
+		fmt.Sprintf("(%s = ?)", searchattribute.GetSqlDbColName(searchattribute.NamespaceID)),
+	)
+	queryArgs = append(queryArgs, namespaceID.String())
+
+	if len(queryString) > 0 {
+		whereClauses = append(whereClauses, fmt.Sprintf("(%s)", queryString))
+	}
+
+	return fmt.Sprintf(
+		"SELECT COUNT(1) FROM executions_visibility WHERE %s",
```
Comment (alexshtin) [inline common/persistence/visibility/store/sql/query_converter_sqlite.go:332]: Had to use ChatGPT to explain me what is `COUNT(1)` and why it is better than `COUNT(*)`.

**Review by alexshtin — APPROVED** [review]

File: common/persistence/visibility/store/sql/query_converter_mysql.go [hunk 2]
```diff
@@ -253,3 +256,32 @@ func (c *mysqlQueryConverter) buildSelectStmt(
...
+	var queryArgs []any
+
+	whereClauses = append(
+		whereClauses,
+		fmt.Sprintf("(%s = ?)", searchattribute.GetSqlDbColName(searchattribute.NamespaceID)),
+	)
+	queryArgs = append(queryArgs, namespaceID.String())
+
+	if len(queryString) > 0 {
+		whereClauses = append(whereClauses, fmt.Sprintf("(%s)", queryString))
```
Comment (yiminc) [inline common/persistence/visibility/store/sql/query_converter_mysql.go:274]: Do we have sql injection risk here?

File: common/persistence/visibility/store/sql/query_converter_mysql.go [hunk 3]
```diff
@@ -253,3 +256,32 @@ func (c *mysqlQueryConverter) buildSelectStmt(
...
+	}
+
+	return fmt.Sprintf(
+		`SELECT COUNT(1)
+		FROM executions_visibility ev
+		LEFT JOIN custom_search_attributes
```
Comment (yiminc) [inline common/persistence/visibility/store/sql/query_converter_mysql.go:280]: Is join here must have? PostgreSQL and SQLite were able to get away without join.

File: common/persistence/visibility/store/sql/query_converter_mysql.go (hunk 2 above)
Comment (rodrigozhou) [reply common/persistence/visibility/store/sql/query_converter_mysql.go:274]: It shouldn't. The query converter escape the user's input.

File: common/persistence/visibility/store/sql/query_converter_mysql.go (hunk 3 above)
Comment (rodrigozhou) [reply common/persistence/visibility/store/sql/query_converter_mysql.go:280]: Yes, in MySQL, we have two tables (executions + custom search attributes), so we need the join.

**Pull request merged**